### ISSUE TITLE
Esprima fails to parse classes that have public fields

### ISSUE THREAD

**Issue by jpstotz** [jquery/esprima#2109]
When trying to parse the example code from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes#public_field_declarations Esprima fails with 
`Error: Line 3: Unexpected token =`

```
var esprima = require('nightly-esprima'); // installed version 2021.8.30

var program = `
class Rectangle {
    height = 0;
    width;
    constructor(height, width) {
      this.height = height;
      this.width = width;
    }
}
`;

esprima.parseScript(program);
```
tested using Node.js v14.18.1

**Comment by jpstotz** [comment]
Looks like one of the waiting pull requests seem to already cover this case, because when I switch to [esprima-next](https://github.com/node-projects/esprima-next) I don't get an error.

**Comment by jogibear9988** [comment]
@jpstotz yes, I fixed this in my fork, cause there seems to be no move forward here...
